engineering/conventions-scala.md

```diff
@@ -0,0 +1,40 @@
+
+# Scala Development Conventions
+
+## Introduction
+
+This guide documents development conventions for Scala at source{d}. Check [general conventions](conventions.md) for language-independent conventions that are also applicable for Scala projects.
+
+## Supported Scala and JVM Versions
+
+Given that most of our Scala projects are related to [Apache Spark](https://spark.apache.org/), our compatibility policy is closely related to it.
+
+* Actively support Java 8. Java 9 on a best-effort basis.
+* Support Scala 2.11.
+
+## Build System
+
+* Use [sbt](https://www.scala-sbt.org/).
+* Include an sbt wrapper in the top-level directory (e.g. [paulp/sbt-extras](https://github.com/paulp/sbt-extras)).
+* Use [sbt-assembly](https://github.com/sbt/sbt-assembly) if fat jars are required.
+* You can use [coursier](https://github.com/coursier/coursier) for faster dependency downloading.
+* Integrate code coverage and scalastyle check in your sbt build files.
+
+## Publishing
+
+* Artifacts should be published to Maven Central under the `tech.sourced` organization.
+
+## Testing
+
+* Use [scalatest](http://www.scalatest.org/).
+* Prefer [FlatSpec](http://www.scalatest.org/user_guide/selecting_a_style) style.
+* Add the `Spec` suffix to test file names.
+
+## Logging
+
+* If working with Spark, you might use `org.apache.spark.internal.Logging`.
+* [slf4j](https://www.slf4j.org/) otherwise.
+
```

Comment:
Shall we also add best practices section, same as [in Go](https://github.com/src-d/guide/pull/196/files#diff-9040191bb0902ff50fda5c0d877fca27R50) and mention  http://twitter.github.io/effectivescala/ ?

Reply:
Added.